posts/MarkoB/2017-02-06-validator-service-loader.adoc

```diff
@@ -0,0 +1,290 @@
+and register them in a specific file in `META-INF/services` directory of your jar. But more on that later.
+
+== Use standard annotations for nonstandard classes
+
+First let's consider a case when you would want one (or few) of standard bean validation
+annotations (or not as standard Hibernate Validator ones) to support some other type, besides
+the ones that are already supported.
+
+=== ThreeTen Extra types validation
+
+So what can be a real life scenario for building your own lib with constraints and sharing it? Well let's say that
+you are building some library with data classes that user might want to validate. As it would be almost
+impossible to keep track of all such libraries and write/maintain all those constraints for them - Hibernate
+Validator provides authors of such libs a possibility to write and share their own validation extensions.
+Which can be picked up by HV and used to validate your data classes.
+
+In this section we will look at http://www.threeten.org/threeten-extra/[ThreeTen Extra types] - a great lib
+that provides additional date-time classes to complement those already present in Java.
+
+As you may already know, Bean Validation (and Hibernate Validator in particular) provides support for the range of
+the https://jcp.org/en/jsr/detail?id=310[JSR 310] date types (javax.time API), by the use of
+`@Past`/`@Future` annotations. So we would want to use these annotations on ThreeTen Extra types as well.
+
+To still keep this example simple we will provide validators only for http://www.threeten.org/threeten-extra/apidocs/org/threeten/extra/DayOfMonth.html[`DayOfMonth`]
+and http://www.threeten.org/threeten-extra/apidocs/org/threeten/extra/Quarter.html[`Quarter`].
+
+We will create a maven project with two modules - one will contain validators, another will contain tests.
+You can look at complete code at http://some-valid-github-link-to-hibernate-demo-repo-will-be-here[github].
+Here's a list of dependencies needed for our validators:
+
+[source,xml]
+._pom.xml_
+<dependencies>
+	<dependency>
+		<groupId>javax.validation</groupId>
+		<artifactId>validation-api</artifactId>
+		<version>2.0.0-SNAPSHOT</version>
+	</dependency>
+	<!--HV dependencies-->
+	<dependency>
+		<groupId>org.hibernate</groupId>
+		<artifactId>hibernate-validator</artifactId>
+		<version>6.0.0-SNAPSHOT</version>
+	</dependency>
+	<dependency>
+		<groupId>org.glassfish</groupId>
+		<artifactId>javax.el</artifactId>
+		<version>3.0.1-b08</version>
+	</dependency>
+	<!--Date-time model dependencies-->
+	<dependency>
+		<groupId>org.threeten</groupId>
+		<artifactId>threeten-extra</artifactId>
+		<version>1.0</version>
+	</dependency>
+	<!--Test dependencies-->
+	<dependency>
+		<groupId>org.testng</groupId>
+		<artifactId>testng</artifactId>
+		<version>6.8</version>
+	</dependency>
+</dependencies>
+
+Validator module does not require whole Hibernate Validator but just validation api (`validation-api`) to get
```

Comment:
right... what I meant here is that you don't need HV for that validator module and validation-api is enough. And by that "whole" part I was saying that "you can use HV as it will bring validation-api as its own dependency, but why do that if you don't need anything from HV ?" 
But as you've said it is not clear so better keep it simple and clean it up :)